edx_notifications/stores/sql/tests/test_store_provider.py

```diff
@@ -94,6 +94,43 @@ def test_save_notification_type(self):
         self.assertIsNotNone(result)
         self.assertEqual(result, notification_type)
 
+    def _save_notification_preference(self, name, display_name, display_description=''):
+        """
+        Helper method to create a new notification_preference
+        """
+        test_notification_preference = NotificationPreference(
+            name=name,
+            display_name=display_name,
+            display_description=display_description
+        )
+
+        test_notification_preference_saved = self.provider.save_notification_preference(test_notification_preference)
```

Comment:
when we typically test these database accessor methods, we wrap these calls in a 'with self.assertNumQueries()' - see above examples - to make sure that the database is only hit N times. This is to catch situations where there might be a performance degradation due to a change. This might take a few times to figure out what the right number of N is (e.g. sometimes create-or-update operations are 2 database calls) for every call site, but you can do:

```
with self.assertNumQueries(1):
    ....some call to self.provider......
```

And then take the number that the assertion raises in the console output and update the assertion count.